core/state_transition.go

```diff
@@ -315,12 +340,39 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 	st.refundGas()
 	st.state.AddBalance(st.evm.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
 
+	if contractCreation && st.evm.SupportsMultitenancy {
```

Comment:
I think we need to remove contractCreation from the test (it may be that an existing contract creates another contract - in which case the transaction has the `to` field populated).

Reply:
Yes, we need to populate the index for all newly created contracts regardless coming from a contract creation tx or message call tx.

Adding